civihr_employee_portal/tests/Helpers/UrlHelperTest.php

```diff
@@ -14,6 +14,63 @@ public function testDedupeWillReturnExpectedResults($url, $expected) {
     $this->assertEquals($expected, UrlHelper::dedupeUrlQueryParams($url));
   }
 
+  /**
+   * @dataProvider urlWithQueryProvider
+   *
+   * @param string $url
+   * @param string $key
+   * @param string $expected
+   */
+  public function testExpectedUrlValuesWillBeRemoved($url, $key, $expected) {
+    $result = UrlHelper::removeQueryValueFromUrl($url, $key);
+    $this->assertEquals($expected, $result);
+  }
+
+  public function urlWithQueryProvider() {
+    return [
+      [
+        'http://example.com?foo=bar',
+        '',
+        'http://example.com?foo=bar',
+      ],
+      [
+        'http://example.com?foo=bar',
+        'foo',
+        'http://example.com',
+      ],
+      [
+        'http://example.com?a=1&b=2',
+        'a',
+        'http://example.com?b=2',
+      ],
+      [
+        'http://example.com?a=1&b=2',
+        'b',
+        'http://example.com?a=1',
+      ],
+      [
+        'http://example.com?a[]=1&a[]=2&b=3',
+        'a',
+        'http://example.com?b=3',
+      ],
+      [
+        'http://example.com?a=1&b[]=2&b[]=3',
+        'b',
+        'http://example.com?a=1',
+      ],
+      [
+        'http://civihr.local/sites/default/files/webform/500.jpg?photo=0',
+        'photo',
+        'http://civihr.local/sites/default/files/webform/500.jpg',
+      ],
+      [
+        'http://example.com?a=1#foo',
```

Comment:
I think this is not even a valid URL, as, according to my understanding, the fragment must be part of the path. I guess it's fine to have a test to show how we expect the helper to work in a situation like that anyway